src/Geometry/CubeGenerator.cpp

```diff
@@ -0,0 +1,1582 @@
+                        5 +
+                    1;
+                elemMPIIndices[((zz * numCubesPerPart[1] + numCubesPerPart[1] - 1) *
+                                    numCubesPerPart[0] +
+                                xx) *
+                                   20 +
+                               7] = nextMPIIndex++;
+                bndLocalIds[nextMPIIndex] =
+                    ((zz * numCubesPerPart[1] + numCubesPerPart[1] - 1) * numCubesPerPart[0] + xx) *
+                        5 +
+                    3;
+                elemMPIIndices[((zz * numCubesPerPart[1] + numCubesPerPart[1] - 1) *
+                                    numCubesPerPart[0] +
+                                xx) *
+                                   20 +
+                               13] = nextMPIIndex++;
+              }
+            }
+          }
+
+          size_t start[3] = {(z * numPartitions[1] + y) * numPartitions[0] + x, bndSize, 0};
+          size_t count[3] = {1, 1, static_cast<unsigned int>(nextMPIIndex)};
+          checkNcError(nc_put_var1_int(ncFile, ncVarBndElemSize, start, &nextMPIIndex));
+          int rank = (z * numPartitions[1] + (y + 1) % numPartitions[1]) * numPartitions[0] + x;
+          rank = (rank + numPartitions[3]) % numPartitions[3];
+          checkNcError(nc_put_var1_int(ncFile, ncVarBndElemRank, start, &rank));
+          checkNcError(nc_put_vara_int(ncFile, ncVarBndElemLocalIds, start, count, bndLocalIds));
+
+          bndSize++;
+        }
+        if ((boundaryMaxz == 6 && numPartitions[2] > 1) ||
+            z != numPartitions[2] - 1) { // last partition in z dimension
+          int nextMPIIndex = 0;
+          for (unsigned int yy = 0; yy < numCubesPerPart[1]; yy++) {
+            for (unsigned int xx = 0; xx < numCubesPerPart[0]; xx++) {
+              //                                                      int odd = (yy+xx+1) % 2;
+              //                                                      if (odd) {
+              bndLocalIds[nextMPIIndex] =
+                  (((numCubesPerPart[2] - 1) * numCubesPerPart[1] + yy) * numCubesPerPart[0] + xx) *
+                      5 +
+                  2;
+              elemMPIIndices[(((numCubesPerPart[2] - 1) * numCubesPerPart[1] + yy) *
+                                  numCubesPerPart[0] +
+                              xx) *
+                                 20 +
+                             11] = nextMPIIndex++;
+              bndLocalIds[nextMPIIndex] =
+                  (((numCubesPerPart[2] - 1) * numCubesPerPart[1] + yy) * numCubesPerPart[0] + xx) *
+                      5 +
+                  3;
+              elemMPIIndices[(((numCubesPerPart[2] - 1) * numCubesPerPart[1] + yy) *
+                                  numCubesPerPart[0] +
+                              xx) *
+                                 20 +
+                             15] = nextMPIIndex++;
+              //                                                      } else {
```

Comment:
Comment still needed?

Reply:
A leftover from the original CubeGenerator. I'll try to figure out if the commented code could be necessary for adjusting the CubeGenerator to read an existing mesh. In the meantime, I will remove the comment for this PR.